build/gulp/tasks/test-projects.tsx

```diff
@@ -15,12 +15,13 @@ const log = msg => {
   console.log('='.repeat(80))
 }
 
-const runIn = path => cmd => sh(`cd ${path} && ${cmd}`)
+export const createPackageFilename = () => tmp.tmpNameSync({ prefix: 'stardust-', postfix: '.tgz' })
 
-const buildAndPackStardust = async (): Promise<string> => {
-  await sh('yarn build:dist')
+export const runIn = path => cmd => sh(`cd ${path} && ${cmd}`)
 
-  return (await sh(`npm pack`, true)).trim()
+export const buildAndPackStardust = async (packageFilename: string) => {
+  await sh('yarn build:dist')
+  await sh(`yarn pack --filename ${packageFilename}`)
```

Comment:
I decided to use `yarn pack` instead of NPM's. We also passing own name, this allows to generate random names and avoid caching locally.

Reply:
I vaguely remember @kuzhelov used `npm` for a reason...

Reply:
the reason for that was just to use the standard tool and not its wrapper, as it could easily entice us to use some features that are `yarn`-specific

Reply:
It looks more cleaner than `(await sh(`npm pack`, true)).trim()`, so I will left `yarn pack` there.